docs/enhancements/1-crds-and-controllers.md

```diff
@@ -0,0 +1,430 @@
+| Status | Authors   | Created    | 
+|--------|-----------|------------|
+| Review | @jewertow | Jan 2 2025 |
+
+# CRDs and controllers
+
+## Overview
+
+Currently, we use helm to manage the controller deployment and its configuration. The configuration is passed to the controller as CLI arguments.
+This approach requires restarting the controller on any configuration change and causes pod failures in case of a misconfiguration.
+
+Additionally, current approach to resource management is not reliable. We apply resources when it's necessary using the Istio client's `Apply` function,
+so we are not able to reconcile created objects on user's modifications or controller uninstallation.
+
+### Goals
+
+Design CRDs for better user experience and ensure proper resource lifecycle and garbage collection.
+
+### Non-goals
+
+Design CRD for installing federation controller. Federation controller will be still installed with Helm,
+and we are not planning any dedicated operator or integration with Sail Operator for now.
+
+## Design
+
+We need the following CRDs:
+1. `MeshFederation` - cluster-scoped resource that includes general federation config, i.e. local settings, remote addresses and identities.
+2. `FederationServiceRules` - specifies rules for exporting and importing services; parent for export-related Istio resources, i.e. `Gateway`, `DestinationRule`, etc.; must be created by the mesh admin.
+3. `ImportedService` - represents an imported service; parent for import-related Istio resources, i.e. `ServiceEntry`, `WorkloadEntry`, etc.; managed only by the controller; should not be touch by users.
+
+All resources will contain [standard conditions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) in their status.
+
+#### MeshFederation
+
+`MeshFederation` includes settings related only to mesh federation, like remote ingress address, identities, etc.
+This resource must be cluster-scoped, because it will be used as an owner for `ImportedService` resources, which can be created in multiple namespaces.
```

Comment:
If that is the only reason to have this resource as cluster-scoped, perhaps we can rethink how we deal with `ImportedService` and their related resources in the context of garbage collection - see https://github.com/openshift-service-mesh/federation/pull/120/files#r1913078586